tools/system_libs.py

```diff
@@ -232,6 +232,23 @@ def create_dlmalloc_split(libname):
     shared.Building.link([dlmalloc_o, split_malloc_o], lib)
     return lib
 
+  def create_wasm_compiler_rt(libname):
+    srcdir = shared.path_from_root('system', 'lib', 'compiler-rt', 'lib', 'builtins')
+    filenames = ['addtf3.c', 'ashlti3.c', 'ashrti3.c', 'comparetf2.c', 'extenddftf2.c', 'fixtfsi.c',
+                 'fixunstfsi.c', 'floatsitf.c', 'floatunsitf.c', 'lshrti3.c', 'multf3.c', 'multi3.c', 'subtf3.c']
+    files = (os.path.join(srcdir, f) for f in filenames)
+    o_s = []
+    commands = []
+    for src in files:
+      o = in_temp(os.path.basename(src) + '.o')
+      commands.append([shared.CLANG_CC, '--target=wasm32', '-S', shared.path_from_root('system', 'lib', src), '-O2', '-o', o])
```

Comment:
instead of calling clang, the others call emcc, which would avoid you needing to write `wasm32` here (which will eventually need to be generalized for `wasm64`, etc.)